### ISSUE TITLE
go-xcat changes to remove kits causes other issues with the tool

### ISSUE THREAD

**Issue by whowutwut** [xcat2/xcat-core#6400]
PR #6385 caused this problem. 

I used go-xcat to install `xcat-version=devel` to get the lastest 2.15...  

Then used go-xcat to uninstall completely, but in this version, build-kit was not removed...

Then using go-xcat again to install without providing a version installed 2.14.6, but now it complains about version mismatch... and partially installed.... This fails forever  because of the validation checks.. 

I found it by running `yum remove` and give it `*xCAT*` , we see the build-kit 2.15  is there with 2.14.6...

```
[root@c910f04x16 xcat]# yum remove "*xCAT*"
Loaded plugins: product-id, search-disabled-repos, subscription-manager
This system is not registered with an entitlement server. You can use subscription-manager to register.
Resolving Dependencies
--> Running transaction check
---> Package perl-xCAT.noarch 4:2.14.6-snap201903290319 will be erased
---> Package xCAT.x86_64 0:2.14.6-snap201903290319 will be erased
---> Package xCAT-buildkit.noarch 4:2.15-snap201908030624 will be erased
---> Package xCAT-client.noarch 4:2.14.6-snap201903290319 will be erased
---> Package xCAT-genesis-base-ppc64.noarch 2:2.14.5-snap201811160710 will be erased
---> Package xCAT-genesis-base-x86_64.noarch 2:2.14.5-snap201811190037 will be erased
---> Package xCAT-genesis-scripts-ppc64.noarch 1:2.14.6-snap201903290319 will be erased
---> Package xCAT-genesis-scripts-x86_64.noarch 1:2.14.6-snap201903290319 will be erased
---> Package xCAT-probe.noarch 4:2.14.6-snap201903290319 will be erased
---> Package xCAT-server.noarch 4:2.14.6-snap201903290319 will be erased
---> Package xCAT-test.noarch 4:2.14.6-snap201903290319 will be erased
--> Finished Dependency Resolution
```

So if we want to use `go-xcat` as a tool to installing/removing and changes xCAT versions, we have fix that bug. 

Seems like we don't have a testcase to catch this, at least supporting `devel` and `ga` ..

**Comment by gurevichmark** [comment]
The PR #6385 caused this problem by removing `xCAT-buildkit` from `GO_XCAT_INSTALL_LIST`. The `GO_XCAT_INSTALL_LIST` is used as part of `GO_XCAT_UNINSTALL_LIST`. As a result, removing xCAT-buildkit` from `GO_XCAT_INSTALL_LIST` caused uninstall to not remove it.

**Comment by besawn** [comment]
Fixed via PR #6402.
